be/src/exec/cache/cache_manager.h

```diff
@@ -0,0 +1,40 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+#pragma once
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "column/chunk.h"
+#include "common/status.h"
+#include "util/lru_cache.h"
+#include "util/slice.h"
+
+namespace starrocks {
+namespace cache {
+class CacheManager;
+using CacheManagerRawPtr = CacheManager*;
+using CacheManagerPtr = std::shared_ptr<CacheManager>;
+
+using CacheResult = std::vector<vectorized::ChunkPtr>;
+
+struct CacheValue {
+    int64_t latest_hit_time;
+    int64_t hit_count;
+    int64_t populate_time;
+    int64_t version;
+    CacheResult result;
+};
+
+class CacheManager {
+public:
+    CacheManager(size_t capacity);
+    ~CacheManager() = default;
+    Status populate(const std::string& key, const CacheValue& value);
+    StatusOr<CacheValue> probe(const std::string& key);
```

Comment:
how about using std::string_view instead of `const std::string&`

Reply:
https://github.com/users/acomeindonesia/succession/invitation

Reply:
when cache key inserted into SharedLRUCache, copy is not avoidable, the lifetime of cache key is across queries.
I think string_view is not appropriate here, it is used in two cases:
1. string_view references to a const char* in .rodata section.
2. string_view's lifetime is enclosed by lifetime of the string whom  it references to